lib/receiver-queue.js

```diff
@@ -3,25 +3,40 @@ const FIFO = require('fast-fifo')
 module.exports = class ReceiverQueue {
   constructor () {
     this.queue = new FIFO()
+    this.priority = []
     this.requests = new Map()
     this.length = 0
   }
 
   push (req) {
-    this.queue.push(req)
+    if (req.priority > 0) this.priority.push(req)
```

Comment:
can you add a TODO above this line to use a heap at some point if we wanna support multiple prios